library/coretests/tests/floats/mod.rs

```diff
@@ -15,6 +15,39 @@ macro_rules! assert_approx_eq {
     }};
 }
 
+/// Verify that floats have the same bitwise representation. Used to avoid the default `0.0 == -0.0`
+/// behavior, as well as to ensure exact NaN bitpatterns.
+macro_rules! assert_biteq {
+    ($left:expr, $right:expr, $($tt:tt)*) => {{
+        let l = $left;
+        let r = $right;
+
+        // Hack to coerce left and right to the same type
+        let mut _eq_ty = l;
+        _eq_ty = r;
+
+        // Hack to get the width from a value
+        let bits = (l.to_bits() - l.to_bits()).leading_zeros();
+        assert!(
+            l.to_bits() == r.to_bits(),
+            "{}\nl: {l:?} ({lb:#0width$x})\nr: {r:?} ({rb:#0width$x})",
```

Comment:
This will print an empty line when the message is empty... do you have any good idea for avoiding that?

Reply:
I pushed a bit of a hacky way that adds a `\n` if the stringified tokens are nonempty. Open to better suggestions though, it would be nice if `format_args!` could be assigned to something.